### PR TITLE
Upload component

### LINKED ISSUE #21: Upgrade submission form
Initially, this will be a form that:
- Allows submission
- Tracks upload progress
- Indicates when successfully received

We'll need the [xhr progress event](https://developer.mozilla.org/en-US/docs/Web/API/ProgressEvent) to track the upload progress, which of course means we'll be using XHR over vanilla form submission.

There is the opportunity to allow for a chunked-upload, where we split the file and only stream over pieces at a time, allowing easy restart if an upload is interrupted, but this requires much more server-side sophistication which is better spent elsewhere (unless we find that interrupted uploads are 1. common and 2. slow enough to warrant this affordance).

### PR DESCRIPTION
**Pull request by wpears**
Basic, implementation of upload component, closes #21

### CODE REVIEW

**Comment by wpears** [comment]
Also closes #3

File: package.json [hunk 1]
```diff
@@ -59,5 +59,9 @@
       "<rootDir>/node_modules/react-addons-test-utils",
       "<rootDir>/node_modules/fbjs"
     ]
+  },
+  "dependencies": {
+    "react": "^0.14.7",
```
Comment (awolfe76) [inline package.json:62]: These being here is right, but should we leave them in `devDependencies` too? Lines 43 and 45.

File: src/js/makeSubmitContainer.jsx [hunk 2]
```diff
@@ -0,0 +1,85 @@
...
+      var className = '';
+      var file = this.props.file;
+      var size = 0;
+      if(!file){
+        className = 'hidden';
+      }else{
+        size = file.size;
+      }
+      return (
+        <div id="uploadTracker" className={className}>
```
Comment (awolfe76) [inline src/js/makeSubmitContainer.jsx]: I like this with the size, but wonder if we should do a progress bar (or something)?
This isn't holding up a merge though, we can always change it later.

File: __tests__/submitContainer.js [hunk 3]
```diff
@@ -1,15 +1,16 @@
...
 var TestUtils = require('react-addons-test-utils');
 
-var FileSubmissionForm = require('../src/js/file-submission-form.jsx');
+jest.autoMockOff();
+
+var makeSubmitContainer = require('../src/js/makeSubmitContainer.jsx');
+var SubmitContainer = makeSubmitContainer();
 
-describe('file-submission-form', function() {
+describe('submitcontainer', function() {
```
Comment (awolfe76) [inline __tests__/submitContainer.js]: Can we add a few tests?
- file selection
- form submission

**Comment by awolfe76** [comment]
Nice! Just a few things there @wpears.

File: src/js/makeSubmitContainer.jsx (hunk 2 above)
Comment (wpears) [inline src/js/makeSubmitContainer.jsx]: Yeah, this will be solved with the PR that addresses https://github.com/cfpb/hmda-platform-ui/issues/34

**Comment by awolfe76** [comment]
:+1:

**Pull request merged**